src/main/java/beam/analysis/plots/RealizedModeAnalysis.java

```diff
@@ -453,12 +453,12 @@ public void writeToReferenceCSV() {
             out.newLine();
 
             StringBuilder builder = new StringBuilder("benchmark");
-            double sum = benchMarkData.values().stream().reduce((x, y) -> x + y).orElse(0.0);
+            double sum = benchMarkData.values().stream().reduce(Double::sum).orElse(0.0);
             for (String d : cumulativeReferenceMode) {
                 if (benchMarkData.get(d) == null) {
                     builder.append(",0.0");
                 } else {
-                    builder.append("," + benchMarkData.get(d) * 100 / sum);
+                    builder.append(",").append(benchMarkData.get(d) * 100 / sum);
```

Comment:
Is this more performant? It seems to me that it would be better to concatenate inside the string?

Reply:
@JustinPihony it is more performant. Concatenating inside the string generates one String object that is appended afterwards and discarded. With the change there is no redundant object creation